src/gengowatcher/notifier.py

```
@@ -2,91 +2,126 @@
     try:
-        subprocess.run(command, check=True)
+        subprocess.run(command, check=True, timeout=10)
         logger.debug("Notification sent successfully via notify-send.")
     except FileNotFoundError:
-        logger.error("`notify-send` command not found. Please ensure it is installed and in your PATH.")
+        logger.error(
+            "`notify-send` command not found. Please ensure it is installed and in your PATH."
+        )
+    except subprocess.TimeoutExpired:
+        logger.warning("notify-send timed out after 10 seconds")
     except subprocess.CalledProcessError as e:
```

Ruff (0.15.2)

[error] Local variable `e` is assigned to but never used

Remove assignment to unused variable `e`

(F841)